CrashReport 3.3.1: java.lang.NullPointerException FileUtils.contentQuery

```
Caused by: java.lang.NullPointerException: 

 
  at android.os.Parcel.createException (Parcel.java:1956)

 
  at android.os.Parcel.readException (Parcel.java:1918)

 
  at android.database.DatabaseUtils.readExceptionFromParcel (DatabaseUtils.java:183)

 
  at android.database.DatabaseUtils.readExceptionFromParcel (DatabaseUtils.java:135)

 
  at android.content.ContentProviderProxy.query (ContentProviderProxy.java:418)

 
  at android.content.ContentResolver.query (ContentResolver.java:809)

 
  at android.content.ContentResolver.query (ContentResolver.java:758)

 
  at android.content.ContentResolver.query (ContentResolver.java:716)

 
  at org.kiwix.kiwixmobile.core.utils.files.FileUtils.contentQuery (FileUtils.java:141)

 
  at org.kiwix.kiwixmobile.core.utils.files.FileUtils.getLocalFilePathByUri (FileUtils.java:114)

 
  at org.kiwix.kiwixmobile.main.KiwixMainActivity.manageExternalLaunchAndRestoringViewState (KiwixMainActivity.java:73)

 
  at org.kiwix.kiwixmobile.main.KiwixMainActivity.onCreate (KiwixMainActivity.java:54)
```

@s-ayush2903 As the screenshots show:

- The crash happens on at least Android 9 and Android 10, it happens in various versions of our app, so it's probably an old bug.

Version|Count|Percent
-- | -- | --
53 |	78|51.7%
182160|45|29.8%
1182160|10|6.6%
6220501|10|6.6%
45|4|2.6%
6220502|3|2.0%
5220501|1|0.7%

- It's happening on various device models.


Liquid S1 (a10) | 22 | 14.6%
-- | -- | --
Galaxy A40 (a40) | 16 | 10.6%
CAMON 12 Air (TECNO-CC6) | 15 | 9.9%
S15 Pro(VISION) (itel-L6002P) | 8 | 5.3%
SPARK 4 (TECNO-KC8) | 6 | 4.0%
Galaxy A10s (a10s) | 5 | 3.3%
Galaxy A50 (a50) | 5 | 3.3%
LT | 5 | 3.3%
Galaxy A20 (a20) | 5 | 3.3%
SPARK 3 (TECNO-KB7j) | 5 | 3.3%
Galaxy S8 (dreamqltecan) | 5 | 3.3%
Smart 3 Plus (Infinix-X627STU) | 4 | 2.6%
Galaxy Note8 (greatqltechn) | 4 | 2.6%
Galaxy S8+ (dream2lte) | 3 | 2.0%
HUAWEI P smart+ 2019 (HWPOT-H) | 3 | 2.0%
Galaxy A50s (a50s) | 3 | 2.0%
Galaxy A60 (a60q) | 3 | 2.0%
Galaxy Note8 (greatlte) | 3 | 2.0%
S15 (itel-W6002E) | 2 | 1.3%
A3_Pro (A3_Pro) | 2 | 1.3%
Others | 27 | 17.9%

Here's an example of the most recent crash's stack trace (the rest are virtually identical)

```
Jul 11, 11:09 PM on app version 53
Tecno CAMON 12 (TECNO-CC7), 3840MB RAM, Android 9
Report 1

java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3013)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3148)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1861)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:193)
  at android.app.ActivityThread.main (ActivityThread.java:6819)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:497)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:912)
Caused by: java.lang.IllegalArgumentException: 
  at android.database.AbstractCursor.getColumnIndexOrThrow (AbstractCursor.java:340)
  at android.database.CursorWrapper.getColumnIndexOrThrow (CursorWrapper.java:87)
  at org.kiwix.kiwixmobile.utils.files.FileUtils.contentQuery (FileUtils.java:165)
  at org.kiwix.kiwixmobile.utils.files.FileUtils.getLocalFilePathByUri (FileUtils.java:150)
  at org.kiwix.kiwixmobile.KiwixMobileActivity.manageExternalLaunchAndRestoringViewState (KiwixMobileActivity.java:1561)
  at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate (KiwixMobileActivity.java:387)
  at android.app.Activity.performCreate (Activity.java:7136)
  at android.app.Activity.performCreate (Activity.java:7127)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1271)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2990)
```

One of the things that does change in the crash report is the line where the crash is triggered e.g. here it's `(FileUtils.java:144)` 

```
Jul 9, 10:32 PM on app version 182160
Infinix Smart 3 Plus (Infinix-X627STU), 2048MB RAM, Android 9
Report 6

java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2956)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3091)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1843)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:193)
  at android.app.ActivityThread.main (ActivityThread.java:6758)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:497)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:912)
Caused by: java.lang.IllegalArgumentException: 
  at android.database.AbstractCursor.getColumnIndexOrThrow (AbstractCursor.java:340)
  at android.database.CursorWrapper.getColumnIndexOrThrow (CursorWrapper.java:87)
  at org.kiwix.kiwixmobile.utils.files.FileUtils.contentQuery (FileUtils.java:159)
  at org.kiwix.kiwixmobile.utils.files.FileUtils.getLocalFilePathByUri (FileUtils.java:144)
  at org.kiwix.kiwixmobile.KiwixMobileActivity.manageExternalLaunchAndRestoringViewState (KiwixMobileActivity.java:1771)
  at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate (KiwixMobileActivity.java:462)
  at android.app.Activity.performCreate (Activity.java:7136)
  at android.app.Activity.performCreate (Activity.java:7127)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1271)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2933)
```
I don't know if this is exactly the same crash as the original one reported, as it may depend on the line number and whether there are several similar lines of the code that could trigger a similar crash. It does *look* similar given `manageExternalLaunchAndRestoringViewState` is in the various crash traces.

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt
@@ -144,6 +144,8 @@ object FileUtils {
         }
     } catch (ignore: SecurityException) {
       null
+    } catch (ignore: NullPointerException) {
+      null
     }
   }
 

